src/main/java/dp/configuration/VaultConfiguration.java

```diff
@@ -0,0 +1,45 @@
+package dp.configuration;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.vault.annotation.VaultPropertySource;
+import org.springframework.vault.annotation.VaultPropertySource.Renewal;
+import org.springframework.vault.authentication.ClientAuthentication;
+import org.springframework.vault.authentication.TokenAuthentication;
+import org.springframework.vault.client.VaultEndpoint;
+import org.springframework.vault.config.AbstractVaultConfiguration;
+
+@Configuration
+public class VaultConfiguration extends AbstractVaultConfiguration {
+
+	@Override
+	public VaultEndpoint vaultEndpoint() {
+		VaultEndpoint vaultEndpoint = null;
+		String addr = System.getenv("VAULT_ADDR");
+		if (addr.length() == 0) {
+			addr = "http://localhost:8200";
+		}
+		try {
+			vaultEndpoint = VaultEndpoint.from(new URL(addr).toURI());
+		} catch (MalformedURLException e) {
+			e.printStackTrace();
+		} catch (URISyntaxException e) {
+			e.printStackTrace();
```

Comment:
Can both exceptions be wrapped in a runtime exception

Reply:
Sorry, I think was not very clear with the comment. Can wrap the old exception and throw it as a run time exception.

eg

```
		} catch (MalformedURLException | URISyntaxException e) {
			throw new RuntimeException(e);
		}
```